src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -619,6 +619,7 @@ private static void DispatchEx(scoped ref StackFrameIterator frameIter, ref ExIn
 
             OnFirstChanceExceptionViaClassLib(exceptionObj);
 
+            uint startIdx = MaxTryRegionIdx;
```

Comment:
Could you please shed some light for me on how the fix works?

As far as I can tell, `puExCollideClauseIdx` argument of `StackIterator.Next` is `out` argument, so passing a different value into `StackIterator.Next` should make difference.

And `isFirstRethrowFrame` should only affect the stacktrace reported by the exception. It should not affect how the exception gets handled.

Reply:
Ah ok, this is for-loop. It means that this `startIdx` is going to be used for the first iteration.